src/Core/Components/Button/FluentCompoundButton.razor.cs

```diff
@@ -0,0 +1,162 @@
+    /// Gets or sets the color of the font (overrides the <see cref="Appearance"/> property).
+    /// </summary>
+    [Parameter]
+    public string? Color { get; set; }
+
+    /// <summary>
+    /// Gets or sets the <see cref="Icon"/> displayed at the start of button content.
+    /// </summary>
+    [Parameter]
+    public Icon? IconStart { get; set; }
+
+    /// <summary>
+    /// Gets or sets the <see cref="Icon"/> displayed at the end of button content.
+    /// </summary>
+    [Parameter]
+    public Icon? IconEnd { get; set; }
+
+    /// <summary>
+    /// Gets or sets the title of the button.
+    /// The text usually displayed in a `tooltip` popup when the mouse is over the button.
+    /// </summary>
+    [Parameter]
+    public string? Title { get; set; }
+
+    /// <summary>
+    /// Gets or sets the element's disabled state, ensuring it doesn't participate in form submission.
+    /// </summary>
+    [Parameter]
+    public bool Disabled { get; set; } = false;
+
+    /// <summary>
+    /// Gets or sets the value indicating the button is focusable.
+    /// </summary>
+    [Parameter]
+    public bool DisabledFocusable { get; set; } = false;
+
+    /// <summary>
+    /// Gets or sets a way to prevent further propagation of the current event in the capturing and bubbling phases.
+    /// </summary>
+    [Parameter]
+    public bool StopPropagation { get; set; } = false;
+
+    /// <summary>
+    /// Gets or sets if the button only shows an icon
+    /// Can be used when using <see cref="ChildContent"/> that renders as an icon
+    /// </summary>
+    [Parameter]
+    public bool IconOnly { get; set; }
+
+    /// <summary>
+    ///  Gets or sets the content to be rendered inside the button.
+    ///  This can be used as an alternative to specifying the content as a child component of the button.
+    ///  If both are specified, both will be rendered.
+    ///  </summary>
+    [Parameter]
+    public string? Label { get; set; }
+
+    /// <summary>
+    /// Gets or sets the content to be rendered inside description area of the button.
+    /// </summary>
+    [Parameter, EditorRequired]
+    public RenderFragment? Description { get; set; }
+
+    /// <summary>
+    /// Gets or sets the content to be rendered inside the button.
+    /// </summary>
+    [Parameter]
+    public RenderFragment? ChildContent { get; set; }
+
+    /// <summary>
+    /// Command executed when the user clicks on the button.
+    /// </summary>
+    [Parameter]
+    public EventCallback<MouseEventArgs> OnClick { get; set; }
+
+    /// <summary />
+    protected async Task OnClickHandlerAsync(MouseEventArgs e)
+    {
+        if (!Disabled && OnClick.HasDelegate)
+        {
+            await OnClick.InvokeAsync(e);
+        }
+
+        await Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Disables the button.
+    /// </summary>
+    /// <param name="disabled">True to disable the button</param>
+    public void SetDisabled(bool disabled)
```

Comment:
I think we need to look to the future and offer only `Async` methods (even if they're not currently useful).

Reply:
Ok, but then this needs to be looked at for Button as well (where I copied it from :))

Reply:
It needs a new PR :-)

i can do that if you prefer.

Reply:
Yes, lets do all of them in a new PR then